lib/cretonne/src/dominator_tree.rs

```diff
@@ -323,6 +329,77 @@ impl DominatorTree {
     }
 }
 
+impl DominatorTree {
+    /// When splitting an `Ebb` using `Layout::split_ebb`, you can use this method to update
+    /// the dominator tree locally rather than recomputing it.
+    ///
+    /// `old_ebb` is the `Ebb` before splitting, and `new_ebb` is the `Ebb` which now contains
+    /// the second half of `old_ebb`. `split_jump_inst` is the terminator jump instruction of
+    /// `old_ebb` that points to `new_ebb`.
+    pub fn recompute_split_ebb(&mut self, old_ebb: Ebb, new_ebb: Ebb, split_jump_inst: Inst) {
+        if !self.is_reachable(old_ebb) {
+            // old_ebb is unreachable, it stays so and new_ebb is unreachable too
+            *self.nodes.ensure(new_ebb) = DomNode {
+                rpo_number: 0,
+                idom: Some(split_jump_inst).into(),
+            };
+        } else {
+            let old_ebb_postorder_index =
+                self.postorder
+            .as_slice()
+            // We use the RPO comparison on the postorder list so we invert the operands of the
+            // comparison
+            .binary_search_by(|probe| self.rpo_cmp_ebb(old_ebb,*probe,))
```

Comment:
What's happening with the commas and spaces here? Strange that rustfmt isn't complaining.

The indentation is also weird, but that could be a GitHub thing.